### ISSUE TITLE
Ansi magenta error

### ISSUE THREAD

**Issue by ViniciusGiroto** [nushell/nushell#7747]
### Describe the bug

Magenta is listed in `ansi -h`, but it raises an error when executed.

### How to reproduce

1. Run `ansi magenta`

### Expected behavior

Print the ansi magenta code. Same as `ansi purple`

### Screenshots

![grafik](https://user-images.githubusercontent.com/9205380/212395569-921559ec-6874-436d-bc63-cfc06f0eccfc.png)

### Configuration

| key                | value                                       |
| ------------------ | ------------------------------------------- |
| version            | 0.74.0                                      |
| branch             |                                             |
| commit_hash        |                                             |
| build_os           | linux-x86_64                                |
| build_target       | x86_64-unknown-linux-gnu                    |
| rust_version       | rustc 1.68.0-nightly (61a415be5 2023-01-12) |
| rust_channel       | nightly-x86_64-unknown-linux-gnu            |
| cargo_version      | cargo 1.68.0-nightly (d992ab4e9 2023-01-10) |
| pkg_version        | 0.74.0                                      |
| build_time         | 2023-01-13 00:22:58 -03:00                  |
| build_rust_channel | release                                     |
| features           | database, default, trash, which, zip        |

### Additional context

`ansi purple` behaviour is the expected of `ansi magenta`. I couldn't find information on why it is called purple and not magenta in Nu.

**Comment by fdncred** [comment]
That's weird, I distinctly remember adding magenta as a synonym for purple. I wonder if that's broken somewhere, either in the nu-ansi-term crate or in the ansi command itself? That shouldn't be too difficult to figure out if someone was so motivated.

**Comment by zschaffer** [comment]
It looks like the HashMap of AnsiCodes doesn't have just 'magenta' as a short or long name. Only 'magenta1', 'magenta2a', 'magenta3' etc 


https://github.com/nushell/nushell/blob/f823c7cb5d0770dfcb75be48be8143c24c01534a/crates/nu-command/src/platform/ansi/ansi_.rs#L359

Testing the original error - it prints fine with `echo $"(magenta1)oi"`

I can't seem to find a place where you'd make a synonym or alias to another AnsiCode besides adding another set of them to CODE_LIST. https://github.com/nushell/nushell/blob/f823c7cb5d0770dfcb75be48be8143c24c01534a/crates/nu-command/src/platform/ansi/ansi_.rs#L20

It looks like purple is properly defined but shouldn't that just be called magenta to conform with most ANSI standards of naming it magenta?

The usage itself lists magenta as the proper color name yet the tests only check for purple 
https://github.com/nushell/nushell/blob/f823c7cb5d0770dfcb75be48be8143c24c01534a/crates/nu-command/src/platform/ansi/ansi_.rs#L537-L546

I'm happy to go in and make the changes needed but unsure what the best way to solve this is. Rename Color::Purple to magenta? Or add a way to covert magenta to purple behind the scenes?

**Comment by fdncred** [comment]
This is from nu-ansi-term and where I remembered making them synonymous
```rust
    /// Color #5 (foreground code `35`, background code `45`).
    Purple,

    /// Color #5 (foreground code `95`, background code `105`).
    LightPurple,

    /// Color #5 (foreground code `35`, background code `45`).
    Magenta,

    /// Color #5 (foreground code `95`, background code `105`).
    LightMagenta,
```
If we're going to add magenta, it just needs to be added along side purple, essentially duplicating this part and making it magenta but making sure none of the names or short names conflict with anything else.
```rust
    AnsiCode{ short_name: Some("p"), long_name: "purple", code: Color::Purple.prefix().to_string()},
    AnsiCode{ short_name: Some("pb"), long_name: "purple_bold", code: Color::Purple.bold().prefix().to_string()},
    AnsiCode{ short_name: Some("pu"), long_name: "purple_underline", code: Color::Purple.underline().prefix().to_string()},
    AnsiCode{ short_name: Some("pi"), long_name: "purple_italic", code: Color::Purple.italic().prefix().to_string()},
    AnsiCode{ short_name: Some("pd"), long_name: "purple_dimmed", code: Color::Purple.dimmed().prefix().to_string()},
    AnsiCode{ short_name: Some("pr"), long_name: "purple_reverse", code: Color::Purple.reverse().prefix().to_string()},
    AnsiCode{ short_name: Some("bg_p"), long_name: "bg_purple", code: Style::new().on(Color::Purple).prefix().to_string()},

    AnsiCode{ short_name: Some("lp"), long_name: "light_purple", code: Color::LightPurple.prefix().to_string()},
    AnsiCode{ short_name: Some("lpb"), long_name: "light_purple_bold", code: Color::LightPurple.bold().prefix().to_string()},
    AnsiCode{ short_name: Some("lpu"), long_name: "light_purple_underline", code: Color::LightPurple.underline().prefix().to_string()},
    AnsiCode{ short_name: Some("lpi"), long_name: "light_purple_italic", code: Color::LightPurple.italic().prefix().to_string()},
    AnsiCode{ short_name: Some("lpd"), long_name: "light_purple_dimmed", code: Color::LightPurple.dimmed().prefix().to_string()},
    AnsiCode{ short_name: Some("lpr"), long_name: "light_purple_reverse", code: Color::LightPurple.reverse().prefix().to_string()},
    AnsiCode{ short_name: Some("bg_lp"), long_name: "bg_light_purple", code: Style::new().on(Color::LightPurple).prefix().to_string()},
```

Then I think `ansi --list` would have to change because you added more colors, so 375 in this line in `generate_ansi_code_list` would have to change.
```rust
            let preview = if i < 375 {
```

**Comment by zschaffer** [comment]
Finished the duplication and updated the ansicode count - looks like magenta and all it's short codes now work as intended

**Comment by fdncred** [comment]
Thanks for working this issue! The changes look good.

**Comment by fdncred** [comment]
closed by #7785
